src/interfaces/chain.cpp

```diff
@@ -4,18 +4,133 @@
 
 #include <interfaces/chain.h>
 
+#include <chain.h>
+#include <chainparams.h>
+#include <primitives/block.h>
 #include <sync.h>
+#include <uint256.h>
 #include <util/system.h>
 #include <validation.h>
 
 #include <memory>
+#include <unordered_map>
```

Comment:
nit: I believe this belongs in `validation.h`, as `unordered_map` isn't used directly here.